tool/util/go.go

```diff
@@ -89,3 +90,50 @@ func SplitCompileCmds(input string) []string {
 func IsGoFile(path string) bool {
 	return strings.HasSuffix(strings.ToLower(path), ".go")
 }
+
+// flagsWithPathValues contains flags that accept a directory or file path as value.
+// From: go help build
+//
+//nolint:gochecknoglobals // constant lookup table
+var flagsWithPathValues = map[string]bool{
+	"-o":       true,
+	"-modfile": true,
+	"-overlay": true,
+	"-pgo":     true,
+	"-pkgdir":  true,
+}
+
+// GetBuildTarget extracts the build target path from the go build command.
+// For example:
+//   - "go build -a cmd/" returns "cmd"
+//   - "go build -a ./app/vmctl" returns "app/vmctl"
+//   - "go build -a ." returns ""
+//   - "go build" returns ""
+//   - "go build -o ./bin/" returns ""
+func GetBuildTarget(goBuildCmd []string) string {
```

Comment:
The go build command supports numerous patterns, which might make exhaustively handling all of them a complex and difficult-to-implement task?

For example, it accepts patterns like:
```
go build cmd
go build std
go build main
go build ./abc/...
go build ./a/. ./b/.
```
Instead of trying to parse these patterns ourselves, perhaps we could leverage golang.org/x/tools/go/packages.Load(). We could pass the patterns (e.g., "cmd", "std", "./abc/...") directly to this function. The packages.Load() function can resolve these patterns and identify the directory D where the go.mod file resides. We could then use this directory D as the working directory for our compilation process?